keras/metrics/metrics.py

```diff
@@ -2645,11 +2646,36 @@ class _IoUBase(base_metric.Metric):
         `(num_classes, num_classes)` will be allocated.
       name: (Optional) string name of the metric instance.
       dtype: (Optional) data type of the metric result.
+      ignore_class: Optional integer. The ID of a class to be ignored during
+        metric computation. This is useful, for example, in segmentation
+        problems featuring a "void" class (commonly -1 or 255) in segmentation
+        maps. By default (`ignore_class=None`), all classes are considered.
+      sparse_labels: Wether labels are encoded using natural numbers or
```

Comment:
Are sparse_labels and sparse_preds actually necessary? Can they not be inferred from the data?

Reply:
I'm surely not the most qualified person to answer this question.

As far as I know, it's impossible to infer this without making assumptions that might render these classes confuse or even useless for end-users in borderline cases. For example:
1. We could assume a dtype `int` means sparse and a dtype `float` means dense. However, `sparse_categorical_crossentropy` works with floats and `categorical_crossentropy` works with ints, so there is no guarantees the user has their label tensor in the right dtype.
2. We could assume sparse labels and dense predictions if `rank(pred) == rank(true) + 1`, and sparse labels and predictions if `rank(pred) == r(true)`; but `sparse_categorical_crossentropy` still works when tensors with completely different shapes/ranks (e.g. 4D predicted output and linearized labels) are given, so there's no guarantees for this case as well.

IMO inferring things generates more headache than it solves, and should be avoided by general-purpose libraries.

Reply:
Ok to keep them, but please rename `sparse_y_true` / `sparse_y_pred` for internal API consistency.